components/webext-storage/sql/create_schema.sql

```diff
@@ -0,0 +1,39 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+-- This is a very simple schema.
+CREATE TABLE IF NOT EXISTS moz_extension_data (
+    ext_id TEXT PRIMARY KEY,
+    /* The JSON payload. not null because we prefer to delete the row (possibly
+       creating a tombstone) than null it */
+    data TEXT NOT NULL,
+
+    /* Same "sync status" strategy used by other components. */
+    syncStatus INTEGER NOT NULL DEFAULT 0,
+    syncChangeCounter INTEGER NOT NULL DEFAULT 1
+) WITHOUT ROWID;
```

Comment:
Excellent question! The sync part of this will use `ext_id` as a foreign key, so I *think* using `ext_id` as the primary key makes sense here - but I'm not sure what the tradeoffs are between what I have, and using a rowid as a primary key with `ext_id` as a non-nullm unique indexed column. I hope Lina chimes in.

Reply:
(to be clear, I think `ext_id` should 100% be the declared primary key on this table; I'm only musing about the totally-invisible-except-for-performance-characteristics things that sqlite might do under the hood given the presence or absence of this `WITHOUT ROWID` declaration).

Reply:
The docs aren't clear - there are parts that make me think that the fact we need the "real" PK to be that string means we are doing the right thing. I need to better understand the btree discussion in paragraph 3 of the doc you linked.